.github/workflows/docs.yml

```diff
@@ -0,0 +1,78 @@
+name: Docs
+
+on:
+  push:
+    branches:
+    - main
+    - 3.0_develop
+  pull_request:
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+    name: Build docs
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+
+    - name: Set up Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: '3.12'
+
+    - name: Install Python dependencies
+      run: |
+        python -m pip install uv
+        uv pip install --system --upgrade pip wheel
+        uv pip install --system --upgrade ".[docs]"
+        uv pip list --system
+
+    - name: Test and build docs
+      run: |
+        pushd docs
+        sphinx-build -b html . _build
+
+    - name: Fix permissions if needed
+      run: |
+        chmod -c -R +rX "docs/_build/" | while read line; do
+          echo "::warning title=Invalid file permissions automatically fixed::$line"
+        done
+
+    - name: Upload artifact
+      uses: actions/upload-pages-artifact@v3
+      with:
+        path: 'docs/_build'
+
+  deploy:
+    name: Deploy docs to GitHub Pages
+    if: github.event_name == 'push' && (github.ref == 'refs/heads/main' || github.ref == 'refs/heads/3.0_develop')
+    needs: build
```

Comment:
> One of the test `Docs / Deploy docs to GitHub Pages` was skipped, how do I enable that?

@ketan96-m Note the dependency here in the `if` statement requires that you're on a branch which is considered a target branch (which needs to change once this huge `3.0_develop` branch goes away) so that you're not deploying with every commit.